examples/placegoose/src/controllers/index.ts

```diff
@@ -0,0 +1,52 @@
+import { eq } from "drizzle-orm";
+import { HTTPException } from "hono/http-exception";
+
+import * as schema from "../db/schema";
+import type { DrizzleClient } from "../types";
+
+// todo: is all this necessary just to check if a record exists?
```

Comment:
existence checks are kind of awkward in drizzle it seems

- https://www.answeroverflow.com/m/1178729503619567628
- https://www.answeroverflow.com/m/1091676597125402644

this seems in part due to the fact you can't have a `select` without a `from` (but that will change soon?)

- https://github.com/drizzle-team/drizzle-orm/pull/1405

so you could do some subquery magic to make a proper existence query, or tick with querying by id and checking length. could also maybe even do a helper that does such an existence check, if you hate repeating the code everywhere 

```ts
function existsById(db: DrizzleClient, table: GenercTypeForSchemaTable, id: number) {
  ...
}
```

Reply:
word; it seems like there are a few options. i'll take a look

Reply:
i've abstracted both the getById and getRowExists logic. wondering whether it's also worth doing one of these:
```
export async function getGaggleById(db: DrizzleClient, id: number) {
  return getRowById(db, schema.gaggles, id);
}
```